src/routes/solid-router/reference/components/memory-router.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: MemoryRouter
+---
+
+The `MemoryRouter` can be used to route while keeping the entire routing history internally in memory.
+Other routers keep the history externally i.e. in the url like the `<HashRouter>` or in the browser history like the [`<Router>`](/solid-router/reference/components/memory-router).
+
+Keeping the history in memory is desirable when you want full control over the routing history.
+The MemoryRouter is therefore often used for testing purposes, where it's controlled through manipulating the `MemoryHistory`.
+
+In the example below, a history object pre-filled with a navigation to the `/about` route is passed to the `MemoryRouter`.
+
+
+
+```jsx
+import { MemoryRouter, createMemoryHistory, A } from "@solidjs/router";
+import { Suspense } from "solid-js";
+
+export default function App() {
+	const history = createMemoryHistory();
+
+	const toHome = () => {
+		history.set({ value: "/" });
+	};
+	const toAbout = () => {
+		history.set({ value: "/about" });
+	};
+
+	return (
+		<>
+			<button onClick={toHome}>{'"/"'}</button>
+			<button onClick={toAbout}>{'"/about"'}</button>
+
+			<MemoryRouter
+				history={history}
+				root={(props) => <Suspense>{props.children}</Suspense>}
+			>
+				{/*... routes */}
+			</MemoryRouter>
+		</>
+	);
+}
+```
+
+## Manipulating the history
+
+The `MemoryHistory` object has the following functions, which are used to control the navigation:
```

Comment:
Can you give a quick blurb about what the functions intend to do? Ie. "To give you control over how you navigate your app" or whatever makes the most sense?

```suggestion
The `MemoryHistory` object contains functions that are used to control navigation. 
```

Reply:
I'm also thinking these are better presented in the form of a table vs a section for each - do you mind putting them into one?

Reply:
This was a great idea.

I added a table, and some text about each of the three types of functions in the MemoryHistory.